src/main/java/com/google/api/codegen/config/SampleSpec.java

```diff
@@ -101,21 +134,26 @@ private static boolean expressionMatchesId(String expression, String id) {
                         .getCallingFormsList()
                         .stream()
                         .anyMatch(expression -> expressionMatchesId(expression, methodFormString)))
-            .flatMap(sampleConfig -> sampleConfig.getValueSetsList().stream())
             .collect(Collectors.toList());
 
-    // Return the value sets corresponding to the selected value set expressions.
-    return valueSets
-        .stream()
-        .filter(
-            valueSet ->
-                valueSetExpressions
-                    .stream()
-                    .anyMatch(expression -> expressionMatchesId(expression, valueSet.getId())))
-        .collect(ImmutableList.toImmutableList());
+    // Construct a `ValueSetAndTags` for each sample specified in each element of `matchingSamples`.
+    List<ValueSetAndTags> result = new ArrayList<>();
+    Set<String> generated = new HashSet<>();
```

Comment:
This seems unused?

Reply:
Good catch! This was cruft from when I was checking for uniqueness here.